clients/include/rocblas_test.hpp

```diff
@@ -118,11 +130,12 @@ bool match_test_category(const Arguments& arg, const char* category);
     INSTANTIATE_TEST_CATEGORY(testclass, known_bug)
 
 // Function to catch signals and exceptions as failures
-void catch_signals_and_exceptions_as_failures(const std::function<void()>& test);
+void catch_signals_and_exceptions_as_failures(const std::function<void()>& test,
+                                              bool                         set_alarm = false);
 
 // Macro to call catch_signals_and_exceptions_as_failures() with a lambda expression
 #define CATCH_SIGNALS_AND_EXCEPTIONS_AS_FAILURES(test) \
-    catch_signals_and_exceptions_as_failures([&] { test; })
+    catch_signals_and_exceptions_as_failures([&] { test; }, true)
```

Comment:
See the alarm going off in CI for gemm_ex so not sure this is expected.  If the process is blocked the alarm doesn't take this into account I don't think.

Reply:
> See the alarm going off in CI for gemm_ex so not sure this is expected. 

If a signal is raised because of a fatal HIP memory error, the alarm might go off on the next test because of deadlocks.

> If the process is blocked the alarm doesn't take this into account I don't think.

The alarm is not activated until a single `gemm_ex` test in `rocblas-test` starts. If the test executable is delayed from starting in the CI, or if there is a long delay between two individual `rocblas-test` tests, then the alarm won't do anything. The alarm is only active between the beginning and ending of a test case. It's possible but rare that a test starts, and then it's suspended by the CI, and then when it wakes up, the alarm fires, but that is unlikely. The alternative of hangs seems much worse.